### ISSUE TITLE
Improve snapshot rollback failure when snapshots produced too frequent.

### ISSUE THREAD

**Issue by puchengy** [apache/iceberg#4144]
Hi all, I am seeing the error below. 
```
java.lang.IllegalArgumentException: [BUG] Expected sorted snapshot log entries.
	at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkArgument(Preconditions.java:142)
	at org.apache.iceberg.TableMetadata.<init>(TableMetadata.java:306)
	at org.apache.iceberg.TableMetadata$Builder.build(TableMetadata.java:1040)
```

Does it mean we shouldn’t write the new snapshot too frequent (within a minute)? I wonder should an improvement be added to the writer side to prevent such error happen? https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/TableMetadata.java#L310

The check is introduced from #1110 #1109

**Comment by wypoon** [comment]
This is a bug introduced in 0.13.0. It is fixed by https://github.com/apache/iceberg/pull/4135 in the 0.13.x branch, and if we release a 0.13.2, it will be there. In master, it is fixed by https://github.com/apache/iceberg/pull/4089.

**Comment by jpugliesi** [comment]
@wypoon thank you for the explanation - is there any known workaround for this?

**Comment by wypoon** [comment]
@jpugliesi there is no workaround for this. However, I believe a 0.13.2 release is planned soon, which will have the fix for this. If you need the fix sooner, you can build Iceberg from the 0.13.x branch yourself.

**Comment by wypoon** [comment]
I suppose if you do need to rollback to a snapshot (and the rollback fails because of this bug), you can create a new table by selecting from the original table as of the snapshot you want, and switch to using the new table.

**Comment by puchengy** [comment]
I can confirm the fix is working for 0.13.x branch. So I am closing this diff. cc: @SinghAsDev
